integration-auth/src/main/java/org/symphonyoss/integration/authentication/AgentApiClientDecorator.java

```diff
@@ -86,18 +86,43 @@
   public AgentApiClientDecorator() {
 
     try {
-      Field defaultHeaderMapField = ApiClient.class.getDeclaredField("defaultHeaderMap");
-      defaultHeaderMapField.setAccessible(true);
-      this.defaultHeaderMap = (Map<String, String>) defaultHeaderMapField.get((ApiClient) this);
-
-      Field authenticationsField = ApiClient.class.getDeclaredField("authentications");
-      authenticationsField.setAccessible(true);
-      this.authentications = (Map<String, Authentication>) authenticationsField.get((ApiClient) this);
+      this.defaultHeaderMap = getDefaultHeaderField();
+      this.authentications = getAuthenticationsField();
     } catch (ReflectiveOperationException e) {
       throw new RuntimeException("Missing parent attributes. Swagger auto-generated code changed and this class requires refactoring.", e);
     }
   }
 
+  /**
+   * Explicity casting to Map generates an unchecked cast warning.
+   * Since the reflection API does not use Generics, this warning can't be avoided.
+   * That's why set the method with SuppressWarnings annotation.
+   * @return Map<String, Authentication>
+   * @throws NoSuchFieldException
+   * @throws IllegalAccessException
+   */
+  @SuppressWarnings("unchecked")
```

Comment:
You could have just put @SuppressWarnings annotation on the original assignment line, right?  With an inline comment of course.